### PR TITLE
OCPBUGS-2083: set TLS cipher suites in Kube RBAC sidecars

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/OCPBUGS-2083
Testing changes for https://github.com/openshift/library-go/pull/1413
The library-go change needs to be merged first and re-vendored in this PR.
/cc @openshift/storage

### CODE REVIEW

**Comment by dobsonj** [comment]
I see some [resize timeouts](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_vmware-vsphere-csi-driver-operator/117/pull-ci-openshift-vmware-vsphere-csi-driver-operator-master-e2e-vsphere-csi/1587458087034294272/artifacts/e2e-vsphere-csi/gather-extra/artifacts/pods/openshift-cluster-csi-drivers_vmware-vsphere-csi-driver-controller-f6b68d994-8t4jr_csi-resizer.log) in e2e-vsphere-csi but it's not obvious to me why:
`type: 'Warning' reason: 'VolumeResizeFailed' resize volume "pvc-77db0318-a6b0-4767-a829-6acd489a03bf" by resizer "csi.vsphere.vmware.com" failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded`
/test e2e-vsphere-csi

File: assets/controller.yaml [hunk 1]
```diff
@@ -228,6 +231,7 @@ spec:
           - --upstream=http://127.0.0.1:8204/
           - --tls-cert-file=/etc/tls/private/tls.crt
           - --tls-private-key-file=/etc/tls/private/tls.key
+          - --tls-cipher-suites=${TLS_CIPHER_SUITES}
```
Comment (gnufied) [inline assets/controller.yaml:234]: iirc - this was an issue with openshift client connecting to vcenter too. We are not going to fix that?

Comment (dobsonj) [reply assets/controller.yaml:234]: Do we have a bug for that somewhere? I don't mind picking it up, I just haven't seen it mentioned before.

Comment (dobsonj) [reply assets/controller.yaml:234]: Also, if this is correct, it looks like they already dropped support for DES ciphers in vSphere 5.x:
https://kb.vmware.com/s/article/1018510
So I don't think we would have to make any client-side changes, but maybe I'm missing something.

**Comment by dobsonj** [comment]
> I see some [resize timeouts](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_vmware-vsphere-csi-driver-operator/117/pull-ci-openshift-vmware-vsphere-csi-driver-operator-master-e2e-vsphere-csi/1587458087034294272/artifacts/e2e-vsphere-csi/gather-extra/artifacts/pods/openshift-cluster-csi-drivers_vmware-vsphere-csi-driver-controller-f6b68d994-8t4jr_csi-resizer.log) in e2e-vsphere-csi

Just looks like infrastructure being slow to respond? From the [csi-driver log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_vmware-vsphere-csi-driver-operator/117/pull-ci-openshift-vmware-vsphere-csi-driver-operator-master-e2e-vsphere-csi/1587553022391619584/artifacts/e2e-vsphere-csi/gather-extra/artifacts/pods/openshift-cluster-csi-drivers_vmware-vsphere-csi-driver-controller-6c56fcf8d-z7shh_csi-driver.log):
`"failed to get taskInfo for ExtendVolume task from vCenter \"ibmvcenter.vmc-ci.devcluster.openshift.com\" with err: Post \"https://ibmvcenter.vmc-ci.devcluster.openshift.com:443/sdk\": context deadline exceeded"`

/retest

**Comment by dobsonj** [comment]
Handshake now fails as expected when requesting 3DES from the client side with:
`openssl s_client -tls1_2 -cipher 3DES -connect 172.31.251.203:9204`
where `172.31.251.203` is one of the master nodes running the csi-resizer container.
Resize also works fine in the VMC environment--CI is failing due to infrastructure issues.

/retitle OCPBUGS-2083: set TLS cipher suites in Kube RBAC sidecars
/retest

**Comment by dobsonj** [comment]
More timeouts even for creating block volumes
`"failed to create volume. Error: Post \"https://ibmvcenter.vmc-ci.devcluster.openshift.com:443/sdk\": context deadline exceeded"`
/retest

**Comment by dobsonj** [comment]
/retest
The best run so far was probably [this one](https://bugzilla.redhat.com/show_bug.cgi?id=2095077), now I see lots of timeouts while provisioning volumes :(

**Comment by gnufied** [comment]
The vsphere-csi test suite is kinda broken right now, until https://issues.redhat.com/browse/OCPBUGS-2927?filter=12399251 is fixed.

**Pull request merged**